File: dev/test_data/main/python/expected/implementation_specific/input/snippets/Verification/has_balanced_brackets.py
def has_balanced_brackets(text: str) -> bool:
    """Check that the square brackets in :paramref:`text` are balanced."""
    depth = 0
    for character in text:
        if character == "[":
            depth += 1
        elif character == "]":
            depth -= 1
            if depth < 0:
                return False

    return depth == 0

File: dev/test_data/main/csharp/expected/implementation_specific/input/snippets/Verification/has_balanced_brackets.cs
/// <summary>
/// Check that the square brackets in <paramref name="text" /> are balanced.
/// </summary>
public static bool HasBalancedBrackets(string text)
{
    int depth = 0;
    foreach (var character in text)
    {
        if (character == '[')
        {
            depth++;
        }
        else if (character == ']')
        {
            depth--;
            if (depth < 0)
            {
                return false;
            }
        }
    }
    return depth == 0;
}
